w3ctrace/state_test.go

```diff
@@ -13,151 +13,144 @@ import (
 	"github.com/instana/testify/require"
 )
 
-const maxKVPairs = 32
-
 func TestParseState(t *testing.T) {
 
 	examples := map[string]struct {
 		Header   string
 		Expected w3ctrace.State
 	}{
 		"empty": {
-			Expected: w3ctrace.NewState([]string{}, ""),
+			Expected: w3ctrace.ParseState(""),
 		},
 		"single tracing system": {
 			Header:   "rojo=00f067aa0ba902b7",
```

Comment:
I understand the intent of removing `NewState` because it is "only" used in tests, however, I think the whole TestParseState test does not make a lot of sense anymore with this change. 

It effectively compares the result of ParseState with another result of ParseState for the same string. In 
https://github.com/instana/go-sensor/pull/312/files#diff-4d31341d37f313670fd6f87a6a018dccc0301c44aab43258fe1f99d8c3341942R100 we create the **actual** value as `ParseState(example.Header)`. Now with this change we also create the **expected** value basically in the same way. We are effectively testing that we use the same string in example.Header as we use in example.Expected (and that ParseState does not return an error).

I think to actually test `ParseState` we need a way to create a `State` struct without relying on `ParseState`.

Reply:
I agree. 👍  I will revert this change.